d2/src/main/java/com/linkedin/d2/xds/GlobCollectionUtils.java

```diff
@@ -1,41 +1,77 @@
 package com.linkedin.d2.xds;
 
+import javax.annotation.Nullable;
+
 public class GlobCollectionUtils
 {
-  private static final String D2_URI_NODE_GLOB_COLLECTION_PREFIX = "xdstp:///indis.D2URI/";
+  private static final String D2_URIS_PREFIX = "/d2/uris/";
+  private static final String D2_URI_NODE_GLOB_COLLECTION_PREFIX = "xdstp:///indis.D2URI/d2/uris/";
   private static final String GLOB_COLLECTION_SUFFIX = "/*";
 
   private GlobCollectionUtils()
   {
   }
 
-  /**
-   * Extracts the glob collection URL (e.g. xdstp:///indis.D2URI/FooCluster/*) from the given resource URN.
-   */
-  public static String globCollectionUrlFromResourceUrn(String resourceUrn)
+  public static class D2UriIdentifier
   {
-    return resourceUrn.substring(0, resourceUrn.lastIndexOf('/')) + GLOB_COLLECTION_SUFFIX;
-  }
+    private final String _clusterPath;
+    private final String _uriName;
 
-  /**
-   * Extracts the URI from the resource URN. Resource URNs for indis.D2URIs have the format of:<pre>
-   *   xdstp:///indis.D2URI/{CLUSTER}/{URL_ESCAPED_ANNOUNCER}
-   * </pre>
-   */
-  public static String uriFromResourceUrn(String resourceUrn)
-  {
-    return resourceUrn.substring(resourceUrn.lastIndexOf('/'));
+    private D2UriIdentifier(String clusterPath, String name)
+    {
+      _clusterPath = clusterPath;
+      _uriName = name;
+    }
+
+    public String getClusterPath()
+    {
+      return _clusterPath;
+    }
+
+    public String getUriName()
+    {
+      return _uriName;
+    }
+
+    /**
+     * Parses the given resource name into a {@link D2UriIdentifier}. URNs for {@code indis.D2URIs} have the format of:
+     * <pre>
+     *   xdstp:///indis.D2URI/{CLUSTER}/{NAME}
+     * </pre>
+     *
+     * @return The parse {@link D2UriIdentifier} if the given resource name is a valid URN, otherwise {@code null}.
+     */
+    @Nullable
+    public static D2UriIdentifier parse(String resourceName)
+    {
+      if (!resourceName.startsWith(D2_URI_NODE_GLOB_COLLECTION_PREFIX))
+      {
+        return null;
+      }
+      int lastIndex = resourceName.lastIndexOf('/');
+      if (lastIndex == 1)
+      {
+        return null;
+      }
+      return new D2UriIdentifier(
+          D2_URIS_PREFIX + resourceName.substring(0, lastIndex) + GLOB_COLLECTION_SUFFIX,
```

Comment:
this will become "/d2/uris/xdstp:///indis.D2URI/d2/uris/{CLUSTER}/*" then?

Reply:
This has been fixed actually, I think you were looking at an old version